cdci_data_analysis/analysis/parameters.py

```diff
@@ -583,6 +583,18 @@ def check_name_value(value, units=None, name=None, par_format=None):
 class Name(String):
     owl_uris = String.owl_uris + ("http://odahub.io/ontology#AstrophysicalObject",)
 
+class FileReference(String):
+    owl_uris = "http://odahub.io/ontology#FileReference"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get_default_value(self):
+        return {'value': self.value, 'downloadable': True}
+
+class POSIXPath(FileReference):
+    owl_uris = FileReference.owl_uris + "http://odahub.io/ontology#POSIXPath"
+
```

Comment:
Not sure, which class must override the `get_default_value()` here. 
This depends on the overall interface. Do we always susbstitute the `POSIXPath` with downloadable file? For the moment, yes.  And what do we do with `FileURL` parameter (not really used yet, but annotation exist)? Do we treat it the same way by uploading from frontend? Or we just pass it as an external URL? 
If both are always "downloadable" and are exposed to the frontend by dispatcher - one `FileReference` class will support both. 
We may discuss a bit in the call

Reply:
> Do we always susbstitute the `POSIXPath` with downloadable file? For the moment, yes. 

Only in the case this is a URL

> And what do we do with `FileURL` parameter (not really used yet, but annotation exist)? Do we treat it the same way by uploading from frontend? 

Is it intended to be a URL provided from the frontend? Will it refer to a File? In that case it might not be needed to be downloadable, so `downloadable=False`

> If both are always "downloadable" and are exposed to the frontend by dispatcher - one `FileReference` class will support both.

With both you mean PosixPAth and FileURL ? Are they always ? I guess it is up to us to decide, but in case they are, one class will suffice

Reply:
Right, that's for us to decide how to provide these urls to the backend (which effectively accepts urls for both). 
Generally speaking, the way frontend + dispatcher deal with files might not be fully determined by the annotations. I could imagine a frontend element which allows to either upload file or provide url, then dispatcher will either expose uploaded file or just pass the url as-is, and it will work for both backend parameter types.
Unfortunately, we didn't really discuss it online. So @volodymyrss what do you think?

Reply:
Can we discuss it on a short meeting? Tomorrow 10h30?

Reply:
Ok for me